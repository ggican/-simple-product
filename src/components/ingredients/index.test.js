import React from "react";
import Ingredients from "./index";
import IngredientsLoading from "./index.loading";
import StyleIngredients from "./index.style";

describe("Ingredients Component", () => {
    let result;

    let successElement;
    let successRecipesElement;
    let failedElement;
    let styleIngredients;
    const onClickIngredient = value => {
        result = value;
    };

    beforeEach(() => {
        styleIngredients = <StyleIngredients></StyleIngredients>;
        successElement = (
            <Ingredients
                dateRecipes="2020-01-12"
                onClickIngredient={onClickIngredient}
                title="Cooking"
                date="2020-12-25"
            ></Ingredients>
        );
        successRecipesElement = (
            <Ingredients
                dateRecipes="2020-01-12"
                onClickIngredient={onClickIngredient}
                isRecipes
                title="Cooking"
            ></Ingredients>
        );

        failedElement = <Ingredients isRecipes={false}></Ingredients>;
    });

    it("SUCCESS renders and create snapshot  <Ingredients />", () => {
        expect(toJson(mount(successElement))).toMatchSnapshot();
    });

    it("SUCCESS renders style and create snapshot  <StyleIngredients />", () => {
        expect(toJson(mount(styleIngredients))).toMatchSnapshot();
    });

    it("SUCCESS renders loading and create snapshot  <IngredientsLoading />", () => {
        expect(
            toJson(
                mount(
                    <IngredientsLoading isRecipes={true}></IngredientsLoading>,
                ),
            ),
        ).toMatchSnapshot();
    });

    it("SUCCESS renders and create snapshot  <Ingredients />", () => {
        expect(toJson(mount(successElement))).toMatchSnapshot();
    });

    it("SUCCESS renders get value title <Ingredients />", () => {
        expect(
            shallow(successElement)
                .find({ test_id: "text-ingredients-top" })
                .find(".ingredients__title")
                .text(),
        ).toEqual("Cooking");
    });

    it("SUCCESS renders get value date <Ingredients />", () => {
        expect(
            shallow(successElement)
                .find({ test_id: "text-ingredients-bottom" })
                .find(".ingredients__date")
                .text(),
        ).toEqual("2020-12-25 ");
    });
    it("SUCCESS renders get value date <Ingredients />", () => {
        expect(
            shallow(successElement)
                .find({ test_id: "text-ingredients-bottom" })
                .find(".ingredients__date")
                .text(),
        ).toEqual("2020-12-25 ");
    });

    it("SUCCESS renders onClick Function <Ingredients />", () => {
        shallow(successElement)
            .find({ test_id: "ingredients" })
            .simulate("click", {
                preventDefault: () => {},
            });
        expect(result).toEqual({ title: "Cooking", date: "2020-12-25" });
    });

    it("SUCCESS renders isDiesabled false <Ingredients />", () => {
        expect(
            shallow(successElement)
                .find({ test_id: "ingredients" })
                .prop("isDisabled"),
        ).toEqual(false);
    });

    it("SUCCESS renders isDisabled false <Ingredients />", () => {
        expect(
            shallow(successElement)
                .find({ test_id: "ingredients" })
                .prop("isDisabled"),
        ).toEqual(false);
    });

    it("SUCCESS renders isRecipes status true <Ingredients />", () => {
        expect(
            shallow(successRecipesElement)
                .find({ test_id: "ingredients" })
                .prop("isRecipes"),
        ).toEqual(true);
    });

    it("SUCCESS renders loading isRecipes true  <IngredientsLoading />", () => {
        expect(
            shallow(
                <IngredientsLoading isRecipes={true}></IngredientsLoading>,
            ).prop("isRecipes"),
        ).toEqual(true);
    });
    it("SUCCESS renders loading isRecipes false  <IngredientsLoading />", () => {
        expect(
            shallow(
                <IngredientsLoading isRecipes={false}></IngredientsLoading>,
            ).prop("isRecipes"),
        ).toEqual(false);
    });

    it("FAILED renders and create snapshot <Ingredients />", () => {
        expect(toJson(mount(successElement))).toMatchSnapshot();
    });

    it("FAILED renders get value title <Ingredients />", () => {
        expect(
            shallow(failedElement)
                .find({ test_id: "text-ingredients-bottom" })
                .find(".ingredients__date")
                .text(),
        ).toEqual("- ");
    });

    it("FAILED renders get value date <Ingredients />", () => {
        expect(
            shallow(failedElement)
                .find({ test_id: "text-ingredients-bottom" })
                .find(".ingredients__date")
                .text(),
        ).toEqual("- ");
    });

    it("FAILED renders get value date <Ingredients />", () => {
        expect(
            shallow(failedElement)
                .find({ test_id: "text-ingredients-bottom" })
                .find(".ingredients__date")
                .text(),
        ).toEqual("- ");
    });
});
