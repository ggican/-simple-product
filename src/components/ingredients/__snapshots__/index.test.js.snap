// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredients Component FAILED renders and create snapshot <Ingredients /> 1`] = `
<Ingredients
  date="2020-12-25"
  dateRecipes="2020-01-12"
  isRecipes={false}
  onClickIngredient={[Function]}
  title="Cooking"
>
  <styled.a
    className="ingredients"
    isDisabled={false}
    isRecipes={false}
    onClick={[Function]}
    test_id="ingredients"
  >
    <a
      className="sc-AxjAm jIXtAR ingredients"
      onClick={[Function]}
    >
      <div
        className="ingredients--box"
      >
        <div
          className="ingredients__text"
          test_id="text-ingredients-top"
        >
          <div
            className="ingredients__caption"
          >
            Title
          </div>
          <div
            className="ingredients__title"
          >
            Cooking
          </div>
        </div>
        <div
          className="ingredients__text"
          test_id="text-ingredients-bottom"
        >
          <div
            className="ingredients__caption"
          >
            Use By
          </div>
          <div
            className="ingredients__date "
          >
            2020-12-25
             
          </div>
        </div>
      </div>
    </a>
  </styled.a>
</Ingredients>
`;

exports[`Ingredients Component SUCCESS renders and create snapshot  <Ingredients /> 1`] = `
<Ingredients
  date="2020-12-25"
  dateRecipes="2020-01-12"
  isRecipes={false}
  onClickIngredient={[Function]}
  title="Cooking"
>
  <styled.a
    className="ingredients"
    isDisabled={false}
    isRecipes={false}
    onClick={[Function]}
    test_id="ingredients"
  >
    <a
      className="sc-AxjAm jIXtAR ingredients"
      onClick={[Function]}
    >
      <div
        className="ingredients--box"
      >
        <div
          className="ingredients__text"
          test_id="text-ingredients-top"
        >
          <div
            className="ingredients__caption"
          >
            Title
          </div>
          <div
            className="ingredients__title"
          >
            Cooking
          </div>
        </div>
        <div
          className="ingredients__text"
          test_id="text-ingredients-bottom"
        >
          <div
            className="ingredients__caption"
          >
            Use By
          </div>
          <div
            className="ingredients__date "
          >
            2020-12-25
             
          </div>
        </div>
      </div>
    </a>
  </styled.a>
</Ingredients>
`;

exports[`Ingredients Component SUCCESS renders and create snapshot  <Ingredients /> 2`] = `
<Ingredients
  date="2020-12-25"
  dateRecipes="2020-01-12"
  isRecipes={false}
  onClickIngredient={[Function]}
  title="Cooking"
>
  <styled.a
    className="ingredients"
    isDisabled={false}
    isRecipes={false}
    onClick={[Function]}
    test_id="ingredients"
  >
    <a
      className="sc-AxjAm jIXtAR ingredients"
      onClick={[Function]}
    >
      <div
        className="ingredients--box"
      >
        <div
          className="ingredients__text"
          test_id="text-ingredients-top"
        >
          <div
            className="ingredients__caption"
          >
            Title
          </div>
          <div
            className="ingredients__title"
          >
            Cooking
          </div>
        </div>
        <div
          className="ingredients__text"
          test_id="text-ingredients-bottom"
        >
          <div
            className="ingredients__caption"
          >
            Use By
          </div>
          <div
            className="ingredients__date "
          >
            2020-12-25
             
          </div>
        </div>
      </div>
    </a>
  </styled.a>
</Ingredients>
`;

exports[`Ingredients Component SUCCESS renders loading and create snapshot  <IngredientsLoading /> 1`] = `
<IngredientsLoading
  isRecipes={true}
>
  <styled.a
    className="ingredients"
    isRecipes={true}
    test_id="ingredients_loading"
  >
    <a
      className="sc-AxjAm bWjhib ingredients"
    >
      <div
        className="ingredients--box"
      >
        <div
          className="ingredients__text"
        >
          <div
            className="ingredients__caption"
          >
            <Skeleton
              height="10px"
              marginBottom="7px"
              width="80px"
            >
              <styled.div
                as="div"
                circle={false}
                height="10px"
                marginBottom="7px"
                style={Object {}}
                width="80px"
              >
                <div
                  className="sc-AxirZ fokiBL"
                  height="10px"
                  style={Object {}}
                  width="80px"
                />
              </styled.div>
            </Skeleton>
          </div>
          <div
            className="ingredients__title"
          >
            <Skeleton
              height="12px"
              marginBottom="7px"
              width="70%"
            >
              <styled.div
                as="div"
                circle={false}
                height="12px"
                marginBottom="7px"
                style={Object {}}
                width="70%"
              >
                <div
                  className="sc-AxirZ beingz"
                  height="12px"
                  style={Object {}}
                  width="70%"
                />
              </styled.div>
            </Skeleton>
          </div>
        </div>
      </div>
    </a>
  </styled.a>
</IngredientsLoading>
`;

exports[`Ingredients Component SUCCESS renders style and create snapshot  <StyleIngredients /> 1`] = `
<styled.a>
  <a
    className="sc-AxjAm jIXtAR"
  />
</styled.a>
`;
